### DISCUSSION TITLE
How to encrypt with multiple GPG keys?

### DISCUSSION THREAD

**Discussion by WhiteBlackGoose** [zeapo/Android-Password-Store#2222] [Q&A]
Hello. Thanks for this great app.
I can't figure out how to encrypt a password with multiple keys simultaneously. I have one private+public key and three public keys on my phone.
When I create a new password, it suggests me to choose one key among the given list from OpenKeyChain. But I'd like to use them all, as I do on my laptop with `pass`

**Comment by msfjarvis** [comment]
Password Store v1 [parses each line of your `.gpg-id`](https://github.com/android-password-store/Android-Password-Store/blob/a2e7d0e5e23639d78eb0652e09cdbadd296341f6/app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt#L342-L368) file as a distinct GPG identifier and forwards them all to OpenKeychain.

If you're being asked to select a key then either your repository does not have a `.gpg-id` file or Password Store was unable to parse any identifiers from it.

**Reply by WhiteBlackGoose** [reply]
I do have `.gpg-id` and it lists all four IDs (btw they all are with the same email, for the record).
However, when I add a new password, it can't be decrypted from my laptop.

FWIW I confirmed the keys with fingerprints in OpenKeyChain but it didn't change anything.

**Reply by WhiteBlackGoose** [reply]
Tried deleting the repo from phone and cloning it again, created a password, OpenKeyChain doens't ask for the public key to use anymore. Still can't be decrypted from laptop (the opposite, laptop -> phone, works)

**Reply by WhiteBlackGoose** [reply]
To double check, I ran `gpg ~/.password-store/file1` and `gpg ~/.password-store/file2`. `file1` is a password I added using phone, and it is encrypted with one key. `file2` is added with `pass`, and it is encrypted with 4 keys

Version of the app: 1.13.5

**Reply by msfjarvis** [reply]
Then it's an OpenKeychain bug which will never be fixed, because the project is no longer maintained 🤷🏼. I already showed you the relevant part of our own code that is doing the correct thing.

**Reply by WhiteBlackGoose** [reply]
I see. Sad. Thanks anyway, good luck with moving away from it!

**Reply by WhiteBlackGoose** [reply]
I found this in your code:
```kt
        return OpenPgpUtils.splitUserId(identifier).email?.let { GpgIdentifier.UserId(it) }
```
in function `parseGpgIdentifier`. If I understand it, it only takes email. All my four IDs have the same email, but they're still different keys. Could it be the issue?

**Reply by WhiteBlackGoose** [reply]
Yep, I tested, works with different emails.
[Reported](https://github.com/android-password-store/Android-Password-Store/issues/2227)
